test-suite/testkit/src/api.rs

```diff
@@ -148,11 +150,58 @@ impl TestKitApi {
             .expect("Cannot broadcast transaction");
     }
 
+    /// Returns the resolved URL for the public API.
+    pub fn public_url(&self, url: &str) -> String {
+        self.test_client.public_url(url)
+    }
+
+    /// Returns the resolved URL for the private API.
+    pub fn private_url(&self, url: &str) -> String {
+        self.test_client.private_url(url)
+    }
+
+    /// Creates a requests builder for the public API scope.
+    pub fn public(&self, kind: impl Display) -> RequestBuilder<'_, '_> {
+        self.test_client.public(kind)
+    }
+
+    /// Creates a requests builder for the private API scope.
+    pub fn private(&self, kind: impl Display) -> RequestBuilder<'_, '_> {
+        self.test_client.private(kind)
+    }
+
+    /// Return reference to the underlying API client.
+    pub fn client(&self) -> &TestKitApiClient {
+        &self.test_client
+    }
+}
+
+/// An asynchronous API client to make Requests with.
+///
+/// This client is wrapper around the `reqwest::Client` to provide more convenient
+/// way to test API.
+#[derive(Debug, Clone)]
+pub struct TestKitApiClient {
+    test_server_url: String,
+    inner: Client,
+}
+
+impl TestKitApiClient {
+    /// Returns the resolved URL for the public API.
+    pub fn public_url(&self, url: &str) -> String {
+        [&self.test_server_url, "public/", url].concat()
```

Comment:
Are you sure that `self.test_server_url` ends with '/' symbol ?

Reply:
Yes.